core/domain/platform_parameter_domain_test.py

```diff
@@ -81,3 +88,1221 @@ def test_to_dict(self):
+        param = parameter_domain.PlatformParameter.from_dict({
+            'name': 'Invalid~Name',
+            'description': 'for test',
+            'data_type': 'string',
+            'rules': [
+                {
+                    'filters': [
+                        {
+                            'type': 'server_mode',
+                            'conditions': [('=', 'dev')]
+                        }
+                    ],
+                    'value_when_matched': '222'
+                }
+            ],
+            'rule_schema_version': (
+                feconf.CURRENT_PLATFORM_PARAMETER_RULE_SCHEMA_VERSION),
+            'default_value': '333',
+            'metadata': {
+                'is_feature': False,
+                'feature_stage': None,
+            },
+        })
+        with self.assertRaisesRegexp(
+            utils.ValidationError, 'Invalid parameter name'):
+            param.validate()
+
+    def test_validate_with_unsupported_data_type_raises_exception(self):
+        param = parameter_domain.PlatformParameter.from_dict({
+            'name': 'parameter_a',
+            'description': 'for test',
+            'data_type': 'InvalidType',
+            'rules': [
+                {
+                    'filters': [
+                        {
+                            'type': 'server_mode',
+                            'conditions': [('=', 'dev')]
+                        }
+                    ],
+                    'value_when_matched': '222'
+                }
+            ],
+            'rule_schema_version': (
+                feconf.CURRENT_PLATFORM_PARAMETER_RULE_SCHEMA_VERSION),
+            'default_value': '333',
+            'metadata': {
+                'is_feature': False,
+                'feature_stage': None,
+            },
+        })
+        with self.assertRaisesRegexp(
+            utils.ValidationError, 'Unsupported data type'):
+            param.validate()
+
+    def test_validate_with_inconsistent_data_type_in_rules_raises_exception(
+            self):
+        param = parameter_domain.PlatformParameter.from_dict({
+            'name': 'parameter_a',
+            'description': 'for test',
+            'data_type': 'bool',
+            'rules': [
+                {
+                    'filters': [
+                        {
+                            'type': 'server_mode',
+                            'conditions': [('=', 'dev')]
+                        }
+                    ],
+                    'value_when_matched': '222'
+                },
+            ],
+            'rule_schema_version': (
+                feconf.CURRENT_PLATFORM_PARAMETER_RULE_SCHEMA_VERSION),
+            'default_value': False,
+            'metadata': {
+                'is_feature': False,
+                'feature_stage': None,
+            },
+        })
+        with self.assertRaisesRegexp(
+            utils.ValidationError, 'Expected bool'):
```

Comment:
Please match the whole error string here.

Reply:
Ditto below

Reply:
Done.